spec/requests/api/v0/user_collections_spec.rb

```diff
@@ -0,0 +1,21 @@
+require "rails_helper"
+
+RSpec.describe "Api::V0::UserCollectionsController", type: :request do
+  before do
+    @user = create(:user)
+    create(:user_collection, user_id: @user.id)
+    create(:user_collection, user_id: @user.id)
+    create(:user_collection, user_id: @user.id)
+  end
+
+  describe "GET /api/user_collections" do
+    it "returns collections for the user" do
+      get '/api/user_collections', params: { user_id: @user.id }
```

Comment:
Need to update test mock the session helper function instead of sending parameter, fetch post now sends sessions information using credentials: include